contracts/facets/ArchaeologistFacet.sol

```diff
@@ -84,4 +86,80 @@ contract ArchaeologistFacet {
     {
         return s.cursedBonds[archaeologist];
     }
+
+    /// @notice Unwraps the sarcophagus.
+    /// @dev Verifies that the unencrypted shard matches the hashedShard stored
+    /// on chain and pays the archaeologist.
+    /// @param identifier The identifier of the sarcophagus
+    /// @param unencryptedShard The archaeologist's corresponding unencrypted shard
+    /// @return The boolean true if the operation was successful
+    function unwrapSarcophagus(
+        bytes32 identifier,
+        bytes memory unencryptedShard
+    ) external returns (bool) {
+        // Confirm that the sarcophagus exists
+        if (
+            s.sarcophaguses[identifier].state !=
+            LibTypes.SarcophagusState.Exists
+        ) {
+            revert LibErrors.SarcophagusDoesNotExist(identifier);
+        }
+
+        // Confirm that the sender is an archaeologist on this sarcophagus
+        if (!LibUtils.archaeologistExistsOnSarc(identifier, msg.sender)) {
+            revert LibErrors.ArchaeologistNotOnSarcophagus(msg.sender);
+        }
+
+        // Confirm that the resurrection time has passed and that the
+        // resurrection window has not passed
+        LibUtils.unwrapTime(
+            s.sarcophaguses[identifier].resurrectionTime,
+            s.sarcophaguses[identifier].resurrectionWindow
+        );
+
+        // Confirm that the archaeologist has not already unwrapped by checking if the unencryptedShard is empty
+        LibUtils.archaeologistUnwrappedCheck(identifier, msg.sender);
+
+        // Comfirm that the sarcophagus has been finalized
+        if (bytes(s.sarcophaguses[identifier].arweaveTxId).length == 0) {
+            revert LibErrors.SarcophagusNotFinalized(identifier);
+        }
```

Comment:
This could potentially get moved to a library (something like `isFinalized(identifier)`), since this logic is used in multiple places.

Reply:
Ignore this comment -- I see you did this in a followup PR